ambry-store/src/main/java/com.github.ambry.store/Journal.java

```diff
@@ -186,4 +188,26 @@ Long getCrcOfKey(StoreKey key) {
   StoreKey getKeyAtOffset(Offset offset) {
     return journal.get(offset);
   }
+
+  /**
+   * Puts the {@link Journal} into bootstrap mode to ignore the {@code maxEntriesToJournal} constraint temporarily.
+   */
+  void startBootstrap() {
+    bootstrap = true;
+  }
+
+  /**
+   * Signals the {@link Journal} is done bootstrapping and will start to honor {@code maxEntriesToJournal}.
+   */
+  void finishBootstrap() {
+    bootstrap = false;
+  }
+
+  /**
+   * @return the number of entries that is currently in the {@link Journal}.
+   */
+  int getCurrentNumberOfEntries() {
```

Comment:
still need this function? for tests?

Reply:
Yes, this function is only needed for testing purpose.